### PR TITLE
remove: updated to match with core repo

### LINKED ISSUE #1490: cmd ref for dvc remove is out of date
1. `dvc remove` description should be updated according to implementation (once  https://github.com/iterative/dvc/issues/4094 is resolved)
1. `--outs` description should be updated according to implementation.
1. `.gitignore` should probably be mentioned/described in examples.

Context:

1. https://github.com/iterative/dvc/pull/4074#issuecomment-647903614
1. https://github.com/iterative/dvc/issues/4094

More problems not mentioned in context above:
1. Extra params mentioned in cmd ref: `-p`, `-f`.
1. There is no shorthand for `--outs` (i.e. `-o`) in the core repo whereas it is mentioned in cmd ref.
1. cmd ref description is also out of date.

### PR DESCRIPTION
**Pull request by nik123**
Fixes #1490 

* `dvc remove` description updated according to implementation (see https://github.com/iterative/dvc/issues/4094 for context)

* `--outs` description updated according to implementation.

* `.gitignore` described in examples.

* Removed extra params: `-p`, `-f`.

* Removed shorthand for `--outs` (i.e. `-o`).

* Updated command description.

### CODE REVIEW

File: content/docs/command-reference/remove.md [hunk 1]
```diff
@@ -1,25 +1,25 @@
 # remove
 
-Remove DVC-tracked files or directories from the <abbr>workspace</abbr>.
+Remove stage entry, remove `.gitignore` entry and unprotect outputs.
```
Comment (jorgeorpinel) [inline content/docs/command-reference/remove.md:3]:
```suggestion
Remove stage, `.gitignore` entry, and unprotect outputs.
```

File: content/docs/command-reference/remove.md [hunk 2]
```diff
@@ -1,25 +1,25 @@
...
 positional arguments:
   targets        stages (found in dvc.yaml) or .dvc files to remove.
 ```
 
 ## Description
 
-This command safely removes data files or directories that are tracked by DVC
-from the <abbr>workspace</abbr>. It takes one or more stage names (see `-n`
-option of `dvc run`) or
+This command safely removes stage entries, `.gitignore` entries and optionally
```
Comment (jorgeorpinel) [inline content/docs/command-reference/remove.md:16]:
```suggestion
This command safely removes stages from `dvc.yaml`, their `.gitignore` entries, and optionally
```
> Paragraph needs formatting.

Comment (jorgeorpinel) [reply content/docs/command-reference/remove.md:16]: What about unprotecting outputs? Not mentioned in the description at all.

File: content/docs/command-reference/remove.md [hunk 3]
```diff
@@ -1,25 +1,25 @@
...
+removes from the <abbr>workspace</abbr> files or directories that are tracked by
+DVC. It takes one or more stage names (see `-n` option of `dvc run`) or
 [`.dvc` files](/doc/user-guide/dvc-files-and-directories#dvc-files) as target,
-removes all of its outputs (outs field), and optionally removes the stage entry
-from [dvc.yaml](/doc/user-guide/dvc-files-and-directories#dvcyaml-file) or the
-`.dvc` file itself.
+removes the stage entry from
```
Comment (jorgeorpinel) [inline content/docs/command-reference/remove.md:20]:
```suggestion
removes it from
```

File: content/docs/command-reference/remove.md [hunk 4]
```diff
@@ -1,25 +1,25 @@
...
+[dvc.yaml](/doc/user-guide/dvc-files-and-directories#dvcyaml-file) or the `.dvc`
+file itself, and optionally removes all of its outputs (outs field).
```
Comment (jorgeorpinel) [inline content/docs/command-reference/remove.md:22]:
```suggestion
removes it, and optionally removes all of its outputs (`outs` field).
```

Comment (jorgeorpinel) [reply content/docs/command-reference/remove.md:22]: What about unprotecting outputs? Not mentioned in the description at all.

File: content/docs/command-reference/remove.md [hunk 5]
```diff
@@ -30,12 +30,7 @@ how it can be used to replace or modify files that are tracked by DVC.
 
 ## Options
 
-- `-o`, `--outs` - remove the outputs described in the given `targets`, keep the
-  `.dvc` files themselves. **This is the default behavior.**
-
-- `-p`, `--purge` - remove outputs and `.dvc` files.
-
-- `-f`, `--force` - force purge. Skip confirmation prompt.
+- `--outs` - remove outputs as well.
```
Comment (jorgeorpinel) [inline content/docs/command-reference/remove.md:33]: Hm let's keep the previous description?
```suggestion
- `--outs` - remove the outputs described in the given `targets`.
```

File: content/docs/command-reference/remove.md [hunk 6]
```diff
@@ -46,14 +41,18 @@ how it can be used to replace or modify files that are tracked by DVC.
...
 ```dvc
 $ dvc add data.csv
 $ ls data.csv*
 
     data.csv
     data.csv.dvc
+
+$ cat .gitignore
+
+    /data.csv
```
Comment (jorgeorpinel) [inline content/docs/command-reference/remove.md:55]: Compact lines, no special indentation please:
```suggestion
$ ls data.csv*
data.csv
data.csv.dvc
$ cat .gitignore
/data.csv
```

File: content/docs/command-reference/remove.md [hunk 7]
```diff
@@ -46,14 +41,18 @@ how it can be used to replace or modify files that are tracked by DVC.
 
 ## Examples
 
-Let's imagine have a `data.csv` data file, and track it with DVC:
+Let's imagine we have a `data.csv` data file, and track it with DVC:
```
Comment (jorgeorpinel) [inline content/docs/command-reference/remove.md:44]:
```suggestion
Let's imagine we have a `data.csv` data file which is already [tracked](link to dvc add cmd ref please) it with DVC:
```
> P needs formatting

File: content/docs/command-reference/remove.md [hunk 8]
```diff
@@ -46,14 +41,18 @@ how it can be used to replace or modify files that are tracked by DVC.
...
 
     data.csv
     data.csv.dvc
+
+$ cat .gitignore
+
+    /data.csv
 ```
 
 Remove `data.csv` data file:
```
Comment (jorgeorpinel) [inline content/docs/command-reference/remove.md:58]:
```suggestion
Remove the `data.csv.dvc` file, and check that the data file is gone from `.gitignore`:
```

File: content/docs/command-reference/remove.md [hunk 9]
```diff
@@ -62,12 +61,9 @@ Remove `data.csv` data file:
 $ dvc remove data.csv.dvc
 $ ls data.csv*
```
Comment (jorgeorpinel) [inline content/docs/command-reference/remove.md:63]:
```suggestion
```

File: content/docs/command-reference/remove.md [hunk 10]
```diff
@@ -62,12 +61,9 @@ Remove `data.csv` data file:
...
 
-     data.csv.dvc
-```
+     data.csv
```
Comment (jorgeorpinel) [inline content/docs/command-reference/remove.md:65]:
```suggestion
data.csv
```

File: content/docs/command-reference/remove.md [hunk 11]
```diff
@@ -62,12 +61,9 @@ Remove `data.csv` data file:
...
 
-Purge `.dvc` files:
+$ cat .gitignore | wc -l
```
Comment (jorgeorpinel) [inline content/docs/command-reference/remove.md:67]:
```suggestion
$ cat .gitignore | wc -l
```

File: content/docs/command-reference/remove.md [hunk 12]
```diff
@@ -62,12 +61,9 @@ Remove `data.csv` data file:
...
 
-```dvc
-$ dvc remove data.csv.dvc -p
-$ ls data.csv*
+    0
```
Comment (jorgeorpinel) [inline content/docs/command-reference/remove.md:68]:
```suggestion
0
```

File: content/docs/command-reference/remove.md (hunk 11 above)
Comment (jorgeorpinel) [reply content/docs/command-reference/remove.md:67]: I'm not sure about this one. It's a little complex, is there a more obvious way to check that .gitignore is empty that doesn't assume people know GNU commands well?

File: content/docs/command-reference/remove.md (hunk 1 above)
Comment (nik123) [reply content/docs/command-reference/remove.md:3]: I thought message should be identical with one in the core repo https://github.com/iterative/dvc/blob/81a4cea71a5a5d54d68575d8c4cacba7ad5563f3/dvc/command/remove.py#L24

I think your suggestion is beitter but probably it should be applied to the core repo first.

File: content/docs/command-reference/remove.md (hunk 5 above)
Comment (nik123) [reply content/docs/command-reference/remove.md:33]: Once again, I used message identical with core repo. Probably `--outs` description should be applied to core repo first,

**Comment by nik123** [comment]
@jorgeorpinel I've applied most of your changes except command text and `--outs` description. I've tried to keep text identical to core repo. I don't know where they should be applied first: here or the core repo.

File: content/docs/command-reference/remove.md (hunk 1 above)
Comment (jorgeorpinel) [reply content/docs/command-reference/remove.md:3]: > I thought message should be identical with one in the core repo

Not here 🙂 just in the code blocks such as under **Synopsis** or command output in examples. No need to apply this exactly to core. Thanks

File: content/docs/command-reference/remove.md (hunk 5 above)
Comment (jorgeorpinel) [reply content/docs/command-reference/remove.md:33]: No don't worry about that, same explanation as in https://github.com/iterative/dvc.org/pull/1497#discussion_r446080399.

**Comment by jorgeorpinel** [comment]
Thanks @nik123!

We only try to keep code blocks identical between core `dvc` command outputs and cmd ref. Not all of the text! So we can safely apply those 2 suggestions.

Do you have any ideas for https://github.com/iterative/dvc.org/pull/1497#pullrequestreview-437182032 ? Again, it's beyond the scope of the PR so no need to do it now, I can do it later. Just lmk please.

File: content/docs/command-reference/remove.md (hunk 11 above)
Comment (nik123) [reply content/docs/command-reference/remove.md]: I have changed the example. Not sure it became better though. Please, check it out.

**Comment by nik123** [comment]
> Do you have any ideas for [#1497 (review)](https://github.com/iterative/dvc.org/pull/1497#pullrequestreview-437182032) ? Again, it's beyond the scope of the PR so no need to do it now, I can do it later. Just lmk please.

As far as understand, there were 2 issues with examples in original code. So I'll try address them both:

**1. Example with dvc-files is too complex**

I've updated example and now it doesn't require reader to know about linux pipes or `wc -l` command. I'm not sure it became better though so please, check it out. May be we'll have to fall back to previous version.

**2. We need "remove stage" example with dvc.yaml file**

Right now I can't come up with nice, short and simple example for dvc.yaml. If you have an idea for one or think that dvc.yaml example shouldn't necessarily be addressed in this PR, I'm totally fine with it.

File: content/docs/command-reference/remove.md (hunk 11 above)
Comment (jorgeorpinel) [reply content/docs/command-reference/remove.md]: It's simpler for sure, that's good. Thanks!

File: content/docs/command-reference/remove.md [hunk 13]
```diff
@@ -46,28 +41,29 @@ how it can be used to replace or modify files that are tracked by DVC.
...
 
-Purge `.dvc` files:
+Remove the `foo.csv.dvc` file, and check that the data file is gone from
+`.gitignore`:
 
 ```dvc
-$ dvc remove data.csv.dvc -p
-$ ls data.csv*
+$ dvc remove foo.csv.dvc
+$ ls *.csv*
```
Comment (jorgeorpinel) [inline content/docs/command-reference/remove.md:63]: Just
```suggestion
$ ls
```

**Comment by jorgeorpinel** [comment]
No worries, I'll address #2.

**Pull request merged**